server/index.js

```diff
@@ -35,6 +39,29 @@ order by
     .catch(err => next(err));
 });
 
+app.get('/api/products/:productId', (req, res, next) => {
+  let productId = parseInt(req.params.productId);
+  productId = [productId];
+  const sql = `
+  select
+  *
+  from
+  "products"
+  where "productId" = $1;
+  `;
+
+  db.query(sql, productId)
+    .then(result => {
+      if (result.rows.length === 0) {
+        next(new ClientError(`Product Id: ${productId} cannot be found`, 404));
```

Comment:
Be sure to `return` here to prevent anything after the `if` from running. We don't want the server to try responding twice.